concrete/src/Http/Middleware/ThumbnailMiddleware.php

```diff
@@ -86,32 +92,77 @@ public function process(Request $request, DelegateInterface $frame)
     }
 
     /**
+     * @param Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    private function buildRedirectToThumbnailResponse(Request $request)
+    {
+        $redirectTo = (string) $request->getUri();
+        $redirectTo .= ((strpos($redirectTo, '?') === false) ? '?' : '&') . mt_rand();
+        $responseFactory = $this->app->make(ResponseFactoryInterface::class);
+        $response = $responseFactory->redirect($redirectTo);
+
+        return $response;
+    }
+
+    /**
+     * @param Connection $database
+     * @param string|null $searchThumbnailPath
+     *
+     * @return ConcretePDOStatement|null
+     */
+    private function getThumbnailPathsToGenerate(Connection $database, $searchThumbnailPath = null)
+    {
+        if ($searchThumbnailPath === null) {
+            $rs = $database->executeQuery('SELECT * FROM FileImageThumbnailPaths WHERE isBuilt = 0 LIMIT 5');
+        } else {
+            $rs = $database->executeQuery('SELECT * FROM FileImageThumbnailPaths WHERE isBuilt = 0 ORDER BY ' . $database->getDatabasePlatform()->getLocateExpression('?', 'path') . ' DESC LIMIT 5', [$searchThumbnailPath]);
+        }
+        if (!$rs->rowCount()) {
+            $rs->closeCursor();
```

Comment:
I don't think you need to call this. This method is meant to be used when you want to iterate over the result set more than one time. Unsetting the result set should be enough.

Honestly, I'd just return the result of executeQuery here.